server/src/lib/model/db/migrations/20191024074951-challenge-related-schemas.ts

```diff
@@ -0,0 +1,77 @@
+export const up = async function(db: any): Promise<any> {
+  return db.runSql(
+    `
+    /* Keep user_clients clean */
+    ALTER TABLE user_clients DROP FOREIGN KEY uc_ibfk_1, DROP COLUMN challenge_team;
+
+    /* A list of challenges. This will only hold one row (the pilot challenge) until at least 2020. */
+    CREATE TABLE challenges (
+      id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+
+      name VARCHAR(255),
+      url_token VARCHAR(36) NOT NULL,
+      start_date DATETIME NOT NULL
+    );
+
+    ALTER TABLE teams DROP PRIMARY KEY,
+              ADD COLUMN id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY, /* better performance and shorter sql statements */
+              CHANGE invite_token url_token VARCHAR(36) NOT NULL,
+              CHANGE team_name name VARCHAR(255),
+              ADD COLUMN challenge_id INT UNSIGNED NOT NULL,                  /* teams has to be associated with exact one challenge to make sense */
+              ADD FOREIGN KEY (challenge_id) REFERENCES challenges(id);
+
+    /* Records each enrollment (a user signing up for a team) across all challenges. */
```

Comment:
Use enrollments instead of enroll is fine.
I was thinking about the table naming convention: a verb means a relation, and an noun mean a entity, But it doesn't show much benefits.

Reply:
Ah, I understand – sorry for missing that.